src/Parameters/JoinMeetingParameters.php

```diff
@@ -131,10 +124,38 @@ class JoinMeetingParameters extends UserDataParameters
      */
     protected $excludeFromDashboard;
 
-    public function __construct(string $meetingID, string $fullName, string $password)
+    public function __construct(string $meetingID, string $fullName, string $password = null)
     {
+        if (\func_num_args() === 3) {
+            @trigger_error(sprintf('Passing $password parameter to constructor of "%s" is deprecated since 5.1 and will be removed in 6.0. Recent BigBlueButton versions does not require the password parameter.', self::class), \E_USER_DEPRECATED);
```

Comment:
Add comment that the role attribute needs to be used instead.
Should we somehow check if either the role or the password is set?

Reply:
I think that validating the co-existence of both parameters is not needed as the password parameters will be gone anyway in the near future. It would be the responsibility of BBB itself to catch such inconsistency and deal with it. In my opinion, logical validation is clearly a responsibility of the API provider and not the API client.